src/Toolkit/Toolkit.Maui/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -47,6 +49,18 @@ static SymbolDisplay()
     public SymbolDisplay()
     {
         ControlTemplate = DefaultControlTemplate;
+        _thisWindowInpcListener = new WeakEventListener<System.ComponentModel.INotifyPropertyChanged, object?, System.ComponentModel.PropertyChangedEventArgs>(this)
+        {
+            OnEventAction = (instance, source, eventArgs) =>
+            {
+                if (eventArgs.PropertyName == nameof(Window) && Window != null && _lastScaleFactor != GetScaleFactor())
+                {
+                    Refresh();
+                }
+            },
```

Comment:
One more issue here. The reference to `Window`, `_lastScaleFactor` and `Refresh` creates hard references to `this`, so this doesn't actually act as a weak reference.  You'll want to get at these through the `instance`/`source` property.
Otherwise looking good